src/PrestaShopBundle/Bridge/Helper/Listing/HelperBridge/HelperListBridge.php

```diff
@@ -157,42 +145,42 @@ protected function generateListQuery(
             'fields' => &$helperListConfiguration->fieldsList,
         ]);
 
-        if (!Validate::isTableOrIdentifier($helperListConfiguration->table)) {
-            throw new PrestaShopException(sprintf('Table name %s is invalid:', $helperListConfiguration->table));
+        if (!Validate::isTableOrIdentifier($helperListConfiguration->getTableName())) {
```

Comment:
`$helperListConfiguration->getTableName()` and `$helperListConfiguration->getListId()` are used many times, consider using variables.

Reply:
Why use an additional variable that uses memory when the value is accessible via the object getter? I don't think it's more performant to use a variable, or is it?

Reply:
by the way its one of places where its mostly copied legacy code, idk if we need to focus on refactoring it

Reply:
It's slightly faster to use a variable rather than calling a function several times, depending on what the function does. The more the function does and the more times you call it, the faster it is to use a variable. But I asked mainly to simplify the code. This can be done later.

I took the time to look of what it takes to repeat a function call, for the fun of it. 

In [this example](https://3v4l.org/5kYI8/vld), calling the simplest of getters takes 5 instructions each time you call it (INIT_METHOD_CALL, DO_FCALL, FETCH_OBJ_R, RETURN, RETURN). Repeating it 3 times makes it 3 * 5 = 15 instructions. 

In [this other example](https://3v4l.org/2d3GA/vld), storing the result of the getter call in a variable assignment takes just 1 operand on top of the 5 to call the getter, then requires no extra instruction for reading it when it is reused. 

Bottom line: if N is the number of instructions to execute a function call, repeating the call X times takes N * X instructions, while storing the result in a variable and reusing it takes just 1 + N instructions. The difference in memory is not really important unless you're storing a very large data set.

Reply:
I have no doubt in performance.
My point is, as you mentioned yourself, we shouldn't be tempted to refactor legacy code. Horizontal migration idea is to provide a bridge and move to symfony as fast as possible, but not to clean up or improve the legacy code. Some other examples - should we refacto every Tool::getValue() usage to use symfony $request? Should we change every Configuration::get() by injecting value in constructor instead? Should we rename legacy variables from snake_case to camelCase or maybe rename them if they sound illogical? I think its not that easy to define the boundaries of optimal refactoring. It is either we require new bridge classes to be fully clean by today standards (and refacto every part of code that is copy-pasted from legacy), or we leave copy-pasted legacy code plain as it is :thinking: